src/CC0-1.0.xml

```diff
@@ -143,6 +143,6 @@
             </list>
         </item>
       </list>
-    
+      <optional><p>For more information, please see &lt;http://creativecommons.org/publicdomain/zero/1.0/&gt;</p></optional>
```

Comment:
Since we don't know where this line comes from (I can't find it anywhere on creativecommons.org, anyway), I'd rather use an `<alt>` to keep it out of the canonical text.  Something like:

```xml
<alt match="For more information, please see <http://creativecommons.org/publicdomain/zero/1.0/>" name="upstreamLink"></alt>

Reply:
It was introduced at https://github.com/github/choosealicense.com/pull/126 with no explanation of that line but I guess it was an attempt to provide something equivalent to the "Back to Commons Deed" link present [at the time](http://web.archive.org/web/20131030222244/https://creativecommons.org/publicdomain/zero/1.0/legalcode). Anyway, you're right, I'll use your suggestion!

Reply:
Hmm, it seems `<p>` still warranted . One way to get it would be to wrap in `<optional>`, but this probably isn't ideal:


```html
      <optional><p><alt match="For more information, please see <http://creativecommons.org/publicdomain/zero/1.0/>" name="upstreamLink"></alt></p></optional>
```

It looks to me as if `<alt match>` hasn't been used with mixed content so far. Can it?

Reply:
Yes - text between ```<alt ...``` and ```</alt>``` will be considered the "original text" and rendered in the web page and any generated text.  You can include formatting for that text.

Reply:
> Hmm, it seems `<p>` still warranted .

Why?  The canonical form has no content there; certainly not an empty paragraph.  And the `match` content need only match the whitespace-normalized license content (per the matching guidelines).  So I don't think we need `<p>` at all.

Reply:
You're right, it didn't occur to me that valid HTML output is irrelevant for text not included in the output. :smile:

Reply:
> … it didn't occur to me that valid HTML output is irrelevant for text not included in the output.

I think there's a larger issue here about “how do we inform folks reading HTML renderings of the set of allowed alternatives?”.  The [current previews][1] just use:

```html
<span class="replacable-license-text">&lt;copyright holder&gt;</span>
```

and similar.  I think we should at least add a `title` attribute with the regexp, although I'm not very excited about exposing the regexps directly to end-users, many of whom may not be familiar with regexps.

[1]: https://spdx.org/licenses/preview/HPND.html